pkg/chunk/cache/memcached_client.go

```diff
@@ -183,6 +196,12 @@ func (c *memcachedClient) Stop() {
 }
 
 func (c *memcachedClient) Set(item *memcache.Item) error {
+	// Skip hitting memcached at all if the item is bigger than the max allowed size.
+	if c.maxItemSize > 0 && uint64(len(item.Value)) > uint64(c.maxItemSize) {
```

Comment:
No need to cast to `uint64`:

```suggestion
	if c.maxItemSize > 0 && len(item.Value) > c.maxItemSize {
```

Reply:
Thanks! Updated.